codegen/type_converter.go

```diff
@@ -73,15 +73,17 @@ type TypeConverter struct {
 	fieldCounter  int
 	convStructMap map[string]string
 	useRecurGen   bool
+	auxChecks     map[string]FieldMapperEntry
 }
 
 // NewTypeConverter returns *TypeConverter
-func NewTypeConverter(h PackageNameResolver) *TypeConverter {
+func NewTypeConverter(h PackageNameResolver, auxChecks map[string]FieldMapperEntry) *TypeConverter {
```

Comment:
`optionalEntries` feels like more indicative to what `auxChecks` does, also let's put a comment for the function argument and explain the intent.